packages/admin/admin-rte/src/core/translation/stateToHtml.spec.ts

```diff
@@ -0,0 +1,511 @@
+                data: {},
+            },
+            {
+                key: "8mn26",
+                text: "Internal News Link",
+                type: "unstyled",
+                depth: 0,
+                inlineStyleRanges: [],
+                entityRanges: [{ offset: 0, length: 18, key: 2 }],
+                data: {},
+            },
+        ];
+        const rawContent = {
+            entityMap,
+            blocks,
+        } as RawDraftContentState;
+
+        const content = convertFromRaw(rawContent);
+        const editorState = EditorState.createWithContent(content);
+
+        const { html, linkDataList } = stateToHtml({
+            editorState,
+            options,
+        });
+
+        const expectedHtml = `<p><a id="0">External Link</a></p>\n<p><a id="1">Internal Link</a></p>\n<p><a id="2">Internal News Link</a></p>`;
+
+        expect(html).toEqual(expectedHtml);
+
+        const expectedLinkDataList = [
+            {
+                id: "0",
+                data: {
+                    activeType: "external",
+                    attachedBlocks: [{ type: "external", props: { targetUrl: "https://www.vivid-planet.com/", openInNewWindow: false } }],
+                },
+            },
+            {
+                id: "1",
+                data: {
+                    activeType: "internal",
+                    attachedBlocks: [
+                        {
+                            type: "internal",
+                            props: {
+                                targetPage: { id: "5aab378c-7a8f-4442-aa33-4aff950d0233", name: "RTE Save", path: "/rte-save", documentType: "Page" },
+                            },
+                        },
+                    ],
+                },
+            },
+            { id: "2", data: { activeType: "news", attachedBlocks: [{ type: "news", props: { id: "3d90dd58-f880-4fd8-a4b8-9ff690655a5b" } }] } },
+        ];
+
+        expect(linkDataList).toEqual(expectedLinkDataList);
+    });
+
+    it("should convert the rte editor state with formating into html while keeping the format via tags - formats part 5 (custom styles)", () => {
+        const blocks = [
+            {
+                key: "7l333",
+                text: "A rte text with custom styling",
+                type: "unstyled",
+                depth: 0,
+                inlineStyleRanges: [{ offset: 0, length: 30, style: "HIGHLIGHT" }],
+                entityRanges: [],
+                data: {},
+            },
+        ];
+
+        const rawContent = {
+            entityMap: {},
+            blocks,
+        } as RawDraftContentState;
+
+        const content = convertFromRaw(rawContent);
+        const editorState = EditorState.createWithContent(content);
+
+        const { html } = stateToHtml({
+            editorState,
+            options,
+        });
+
+        const expectedHtml = `<p><span style="HIGHLIGHT">A rte text with custom styling</span></p>`;
```

Comment:
This is not correct HTML please use `class` instead of `style` for the custom styles